src/sidebar/services/groups.js

```diff
@@ -211,6 +235,26 @@ export class GroupsService {
     return this._api.group.read({ id, expand: expandParam });
   }
 
+  /**
+   * Load a subset of the user's full set of groups
+   *
+   * @param {string[]} groupIDs - IDs of groups we want to use
+   * @param {Group[]} allGroups - Full set of user's groups from profile
+   */
+  _loadFilteredGroups(groupIDs, allGroups) {
+    const filteredGroups = groupIDs.map(id =>
+      allGroups.find(g => g.id === id || g.groupid === id)
+    );
+    // In case any groups were not found
+    const groups = /** @type {Group[]} */ (filteredGroups.filter(g => !!g));
+
+    // Restore focused group if possible (won't work if previous focused group
+    // is not in the filtered subset)
+    const focusedGroupId = this._store.directLinkedGroupId();
+
+    this._addGroupsToStore(groups, focusedGroupId);
```

Comment:
The current approach to filtering groups in the store involves changing the set of groups that are actually loaded into the store.

An alternative might be to make the various selectors in the store which return groups reflect this filter. If this works, it may remove the need for the service-level logic. A quick sketch:

```diff
diff --git a/src/sidebar/store/modules/groups.js b/src/sidebar/store/modules/groups.js
index 2c56001f4..ec492af34 100644
--- a/src/sidebar/store/modules/groups.js
+++ b/src/sidebar/store/modules/groups.js
@@ -149,7 +149,10 @@ function focusedGroupId(state) {
  * @return {Group[]}
  */
 function allGroups(state) {
-  return state.groups;
+  if (state.filteredGroupIDs.length === 0) {
+    return state.groups;
+  }
+  return state.groups.filter(g => state.filteredGroupIDs.includes(g.id));
 }
 
 /**
@@ -167,7 +170,7 @@ function getGroup(state, id) {
  */
 const getFeaturedGroups = createSelector(
   /** @param {State} state */
-  state => state.groups,
+  state => allGroups(state),
   groups => groups.filter(group => !group.isMember && group.isScopedToUri)
 );
 
@@ -178,7 +181,7 @@ const getFeaturedGroups = createSelector(
  */
 const getInScopeGroups = createSelector(
   /** @param {State} state */
-  state => state.groups,
+  state => allGroups(state),
   groups => groups.filter(g => g.isScopedToUri)
 );
 
@@ -189,7 +192,7 @@ const getInScopeGroups = createSelector(
  */
 const getMyGroups = createSelector(
   /** @param {{ groups: State, session: SessionState }} rootState */
-  rootState => rootState.groups.groups,
+  rootState => allGroups(rootState.groups),
   rootState => session.selectors.isLoggedIn(rootState.session),
   (groups, loggedIn) => {
     // If logged out, the Public group still has isMember set to true so only
```